### PR TITLE
Add python wrapper for gather op.

### PR DESCRIPTION
**Pull request by wanghaoshuang**
Add python wrapper for gather op.

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -3978,3 +3979,53 @@ def _is_list_or_turple_(data):
...
+                Index = [1, 2]
+
+                Then:
+
+                Out = [[3, 4],
+                       [5, 6]]
+
+    Args:
+        input (Variable): The source input with rank>1. 
+        index (Variable): The index input with rang=1.
```
Comment (baiyfbupt) [inline python/paddle/fluid/layers/nn.py]: rank?

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -3978,3 +3979,53 @@ def _is_list_or_turple_(data):
...
                "out_w": out_w})
     return out
+
+
+def gather(input, index):
+    """
+    **Gather Operator**
+
+    Out is obtained by gathering entries of the outer-most dimension 
+    of X indexed by Index and concatenate them together.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: 1. Out -> Output
2. 
```
Index  -> `index`
```

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -3978,3 +3979,53 @@ def _is_list_or_turple_(data):
...
+
+    .. math::
+
+	Out = X[Index]
+
+
+    .. code-block:: text
+
+	* Case 1:
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: 还有其他Case吗？否则，就不要标Case了

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -3978,3 +3979,53 @@ def _is_list_or_turple_(data):
...
+
+                Out = [[3, 4],
+                       [5, 6]]
+
+    Args:
+        input (Variable): The source input with rank>1. 
+        index (Variable): The index input with rang=1.
+
+    Returns:
+        output (Variable): The output is a tensor with the same shape as input.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: 输出和输入shape并不相同。

File: python/paddle/fluid/layers/nn.py [hunk 5]
```diff
@@ -3978,3 +3979,53 @@ def _is_list_or_turple_(data):
         attrs={"out_h": out_h,
                "out_w": out_w})
     return out
+
+
+def gather(input, index):
+    """
+    **Gather Operator**
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: In Python API, there should not be any operator.

File: python/paddle/fluid/layers/nn.py [hunk 6]
```diff
@@ -3978,3 +3979,53 @@ def _is_list_or_turple_(data):
...
+
+                Index = [1, 2]
+
+                Then:
+
+                Out = [[3, 4],
+                       [5, 6]]
+
+    Args:
+        input (Variable): The source input with rank>1.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: gather_op should work even if `rank == 1`, please consider to modify it.

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (wanghaoshuang) [reply python/paddle/fluid/layers/nn.py]: Thx. Fixed.

File: python/paddle/fluid/layers/nn.py (hunk 5 above)
Comment (wanghaoshuang) [reply python/paddle/fluid/layers/nn.py]: Thx. Fixed.

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (wanghaoshuang) [reply python/paddle/fluid/layers/nn.py]: Thx. Fixed.

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (wanghaoshuang) [reply python/paddle/fluid/layers/nn.py]: Thx. Fixed.

File: python/paddle/fluid/layers/nn.py (hunk 6 above)
Comment (wanghaoshuang) [reply python/paddle/fluid/layers/nn.py]: Thx. The op support `rank==1` already. I add a unitest for `rank==1`.

**Pull request merged**